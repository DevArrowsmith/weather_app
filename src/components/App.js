import React, { useState, useEffect } from "react";
import "../styles/App.css";
import SearchForm from "./SearchForm";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";

const App = () => {
    const [locationSearchText, setLocationSearchText] = useState("");
    const [forecasts, setForecasts] = useState([]);
    const [location, setLocation] = useState({ city: "", country: "" });
    const [selectedDate, setSelectedDate] = useState(0);
    useEffect(() => {
        getForecast(
            locationSearchText,
            setSelectedDate,
            setForecasts,
            setLocation
        );
    }, []);

    const handleCitySearch = () => {
        getForecast(
            locationSearchText,
            setSelectedDate,
            setForecasts,
            setLocation
        );
    };

    const handleForecastSelect = (date) => {
        setSelectedDate(date);
    };

    const selectedForecast = forecasts.find(
        (forecast) => forecast.date === selectedDate
    );

    return (
        <div className="weather-app">
            <h1>Weather App</h1>
            <SearchForm
                searchText={locationSearchText}
                setSearchText={setLocationSearchText}
                onSubmit={handleCitySearch}
            />
            <LocationDetails city={location.city} country={location.country} />
            {selectedForecast && (
                <ForecastDetails forecast={selectedForecast} />
            )}
            <ForecastSummaries
                forecasts={forecasts}
                onForecastSelect={handleForecastSelect}
            />
        </div>
    );
};

export default App;
