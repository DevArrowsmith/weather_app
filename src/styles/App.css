.weather-app {
    width: 100vw;
    height: 100vh;
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    background-color: #EBEBEB;
};
