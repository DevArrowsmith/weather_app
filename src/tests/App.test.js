import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";
import { location, forecasts } from "../data/mock-forecast.json";

describe("App component", () => {
    test("Renders correctly (snapshot)", () => {
        const { asFragment } = render(
            <App forecasts={forecasts} location={location} />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test("renders Weather App title", () => {
        render(<App forecasts={forecasts} location={location} />);
        const titleElement = screen.getByText(/Weather App/i);
        expect(titleElement).toBeInTheDocument();
    });
});
