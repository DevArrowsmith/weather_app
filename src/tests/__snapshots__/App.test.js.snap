// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Renders correctly (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      Weather App
    </h1>
    <div
      class="search-form"
    >
      <input
        placeholder="Location"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </div>
    <h1>
      , 
    </h1>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
